scripts/split_identify_fasta.py

```diff
@@ -11,35 +11,44 @@
 parser.add_argument("--Ns", type = int, default = 5000, help = "Number of Ns to split by [5000]")
 
 args = parser.parse_args()
+def read_fasta(fp):
+    name, seq = None, []
+    for line in fp:
+        line = line.rstrip()
+        if line.startswith(">"):
+            if name: yield (name, ''.join(seq))
+            name, seq = line, []
+        else:
+            seq.append(line)
+    if name: yield (name, ''.join(seq))
+     
 
 with open(args.outputfile, "w+") as outf:
     with open(args.inputfile) as f:
-        for linenr, line in enumerate(f):
-            if linenr % 2 == 0:
-                rid = line.rstrip().lstrip(">")
-            else:
-                if line[0] not in {'A', 'G', 'C', 'T', 'N'}:
-                    print("Error found in input file. Make sure the input is in correct fasta format.")
-                    sys.exit()
-                print(rid + "... ", end = "")
-
-                s = line.rstrip().upper().lstrip("N").rstrip("N")
-                matches = re.finditer("N{" + str(args.Ns) + ",}", s)
-                start = 0
-                partnr = 1
-                for m in matches:
-                    nstart, nstop = m.span()
-                    #print(s[m.span()[1]])
-                    part = s[start:nstart]
-
-                    outf.write(">" + args.prefix + "_" + rid + "_" + str(partnr) + "\n")
-                    outf.write(part + "\n")
-
-                    partnr += 1
-                    start = nstop
-                part = s[start:]
-                outf.write(">" + args.prefix + "_" + rid + "_" + str(partnr) + "\n")
+        for fid, seqt in read_fasta(f):
+            seq = seqt.upper()
+            if seq[0] not in {'A', 'G', 'C', 'T', 'N'}:
```

Comment:
This is true for human assemblies---AGCTN. 

The issue came up for plants and viruses that these FASTAs commonly contain e.g. R, Y, K, M. There's also U. 

A more informative check might be useful---there are other ways to check that the FASTA is the correct format.

Reply:
I added more letters. This is just a quick check however, I don't want to do a comprehensive check if this is proper fasta. Somebody using the pipeline should have some prior knowledge of how fasta looks like.

Reply:
> Somebody using the pipeline should have some prior knowledge of how fasta looks like.

I agree!

I could see this as an annoying snag though, which drives bioinformaticians crazy....these edge cases which you spend your days trying to figure out :)